src/charm.py

```diff
@@ -88,27 +87,34 @@ def _on_config_changed(self, _: ops.ConfigChangedEvent) -> None:
         self.unit.status = ops.ActiveStatus()  # pragma: no cover
 
     @charm_utils.block_if_invalid_config(defer=False)
-    def _on_image_relation_changed(self, _: ops.RelationChangedEvent) -> None:
+    def _on_image_relation_changed(self, evt: ops.RelationChangedEvent) -> None:
```

Comment:
I believe this fct should be moved to the Observer. Did not tackle this in this PR.